src/usb.rs

```diff
@@ -24,11 +24,12 @@ unsafe impl UsbPeripheral for Peripheral {
     const DP_PULL_UP_FEATURE: bool = false;
     const EP_MEMORY: *const () = 0x4000_6000 as _;
     const EP_MEMORY_SIZE: usize = 512;
+    const EP_MEMORY_ACCESS_2X16: bool = false;
 
     fn enable() {
-        let rcc = unsafe { (&*RCC::ptr()) };
+        cortex_m::interrupt::free(|_| unsafe {
```

Comment:
Will this not make the `unsafe` region larger than nessecary? It's only required for the ptr, right?

Reply:
As far as I understand unsafety, the unsafe operation here is accessing the memory via a pointer constructed in unsafe way, not just constructing the pointer. So larger unsafe region looks more correct compared to what we had before.

Reply:
No, the unsafe code create a reference, not a pointer. So creating this reference is hightly unsafe, but using the reference is not.

Reply:
Creating this reference is better inside the `interrupt::free`. So I propose to just move the `let rcc` line inside the interrupt::free.

Reply:
With #344 be merged we don't need critical section as it uses bit-banding.